kratos/sources/oriented_bounding_box.cpp

```diff
@@ -82,22 +84,57 @@ OrientedBoundingBox<2>::OrientedBoundingBox(
 template<>
 OrientedBoundingBox<3>::OrientedBoundingBox(
     const GeometryType& rGeometry,
-    const double BoundingBoxFactor
+    const double BoundingBoxFactor,
+    const bool BuildFromBoundingBox
     )
 {
-    // Check the intersection of each face of the object bounding box against the intersecting object bounding box
-    NodeType geometry_low_node, geometry_high_node;
-    rGeometry.BoundingBox(geometry_low_node, geometry_high_node);
+    if (BuildFromBoundingBox || rGeometry.WorkingSpaceDimension() == rGeometry.LocalSpaceDimension()) {
+        // Check the intersection of each face of the object bounding box against the intersecting object bounding box
+        NodeType geometry_low_node, geometry_high_node;
+        rGeometry.BoundingBox(geometry_low_node, geometry_high_node);
+
+        // Creating OBB
+        noalias(mOrientationVectors[0]) = geometry_high_node - geometry_low_node;
+        const double norm_mOrientationVectors = norm_2(mOrientationVectors[0]);
+        if (norm_mOrientationVectors > std::numeric_limits<double>::epsilon())
```

Comment:
I'm wondering if the `std::numeric_limits<double>::epsilon()` could lead to valid but ill-conditioned cases.

Reply:
This is already working...

Reply:
I willl define a static value, wo if we need to update only is needed once